hudi-common/src/main/java/org/apache/hudi/common/table/read/FileGroupRecordBuffer.java

```diff
@@ -394,4 +372,29 @@ static Comparable getOrderingValue(HoodieReaderContext readerContext,
         ? DEFAULT_ORDERING_VALUE
         : readerContext.convertValueToEngineType(deleteRecord.getOrderingValue());
   }
+
+  private static class LogRecordIterator<T> implements ClosableIterator<BufferedRecord<T>> {
+    private final FileGroupRecordBuffer<T> fileGroupRecordBuffer;
+    private final Iterator<BufferedRecord<T>> logRecordIterator;
+
+    private LogRecordIterator(FileGroupRecordBuffer<T> fileGroupRecordBuffer) {
+      this.fileGroupRecordBuffer = fileGroupRecordBuffer;
+      this.logRecordIterator = fileGroupRecordBuffer.records.values().iterator();
```

Comment:
Regarding the OOM comment I have on `HoodieTableMetadataUtil`, it can be fixed by using `fileGroupRecordBuffer.records.iterator()` instead of `fileGroupRecordBuffer.records.values().iterator()`.

Reply:
Thanks for catching this, I updated it